### PR TITLE
1051 rh UI re work first steps

### PR DESCRIPTION
**Pull request by kacperpONS**
# Motivation and Context

First steps for creating the new rh-ui project.

# What has changed

I've created the basic project structure, added a basic Flask app, with a Hello World page and some simple tests.
I've added the Dockerfile, set up logging, flake8 and Vulture.

# How to test?

Check that all of the works and makes sense.
Check that the repo settings match our guidelines.
Check that the `make` commands work and make sense.
Make sure we're happy with the structure of the directories in the project.

# Links

https://trello.com/c/Tj9FLkF3/1051-rh-ui-re-work-first-steps-8

# Screenshots (if appropriate):

### CODE REVIEW

**Comment by kacperpONS** [comment]
Work in progress: I'm trying to set up the GH Actions secrets.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,36 @@
...
+Then (and this is a change from the ras ui pattern explained below) we register the blueprints against our Flask app in the `create_app` function
+
+# Questions
+## How will we go about defining app configs? 
+It seems that RAS UI picks up the app config through env variables, but at the same time in each config they add a boolean variable that describes what the config is supposed to be (TESTING, DEVELOPMENT etc.)
+To do: investigate how that was done in the old RH-UI, decide what works for us better
+
+# Comments
+
+## Global `app` object
```
Comment (ryangrundy7) [inline README.md:20]: Weird space character here

File: README.md [hunk 2]
```diff
@@ -1,2 +1,36 @@
...
+It seems that RASRM create the app in the __init__.py file making it avaiable in the whole project
+I didn't see any benefit in that for us, and it entails some complications (for example in tests), so I decided to leave the app creation in the run.py script 
+All the app configuration happens in the `create_app` function instead
+
+## Error blueprints
```
Comment (ryangrundy7) [inline README.md:25]: Weird space character here

File: README.md [hunk 3]
```diff
@@ -1,2 +1,36 @@
...
+In the views folder we define our view functions as Flask Blueprints (following the pattern from ras-frontstage)
+
+Then (and this is a change from the ras ui pattern explained below) we register the blueprints against our Flask app in the `create_app` function
+
+# Questions
+## How will we go about defining app configs? 
+It seems that RAS UI picks up the app config through env variables, but at the same time in each config they add a boolean variable that describes what the config is supposed to be (TESTING, DEVELOPMENT etc.)
+To do: investigate how that was done in the old RH-UI, decide what works for us better
+
+# Comments
```
Comment (ryangrundy7) [inline README.md:18]: Weird space character here

File: rh_ui/app_setup.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+from flask import Flask
+from structlog import wrap_logger
+from rh_ui.logger_config import logger_initial_config
+
+
+def create_app():
+    app = Flask("RH-UI app")
+
+    app_config = "config.{}".format(os.environ.get("APP_CONFIG", "DevelopmentConfig"))
```
Comment (ryangrundy7) [inline rh_ui/app_setup.py:12]: F-strings ftw :art: 
```suggestion
    app_config = f'config.{os.environ.get("APP_CONFIG", "DevelopmentConfig")}'
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,31 @@
+run:
+	pipenv run python run.py
+
+install:
+	pipenv install --dev
+
+load_templates:
+	./scripts/load_templates.sh
```
Comment (ryangrundy7) [inline Makefile:8]: No scripts directory, is this command needed?

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+flake8:
+	pipenv run flake8 --exclude=whitelist.py 
+
+vulture:
+	pipenv run vulture .
+
+update_vulture_whitelist:
+	pipenv run vulture . --make-whitelist > whitelist.py
+
```
Comment (ryangrundy7) [inline Makefile:27]: Command works but I see this error
```
ubuntu-2004-022 :: ~/projects/srm-rh-ui ‹1051-rh-ui-re-work-first-steps*› » make update_vulture_whitelist                       
pipenv run vulture . --make-whitelist > whitelist.py
make: *** [Makefile:26: update_vulture_whitelist] Error 1
```

File: Makefile
Comment (ryangrundy7) [inline Makefile:1]: We usually have all the repos have this make command as well
```
build: unit_tests docker-build
```

File: tests/unit/views/test_hello.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+def test_cookies_success(test_client):
```
Comment (ryangrundy7) [inline tests/unit/views/test_hello.py:1]: Should be renamed to `test_hello_endpoint_success` or something similar?

File: Makefile (hunk 6 above)
Comment (kacperpONS) [reply Makefile]: It seems that the vulture command for some reason returns an error code 1 (even though it works exactly as expected) and then the Makefile picks it up as an error. As a workaround we've agreed to just add `|| true` at the end of the command.

**Review by ryangrundy7 — APPROVED** [review]
Good starting point for new RH-UI

**Review by hbrace — APPROVED** [review]
Looks like a good start 👍

**Pull request merged**